### PR TITLE
bpo-40956: Convert _sqlite3.Connection to Argument Clinic, part 2

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue40956

### CODE REVIEW

**Comment by erlend-aasland** [comment]
@corona10, here's the rest of `_sqlite3.Connection`.

**Review by ghost — APPROVED** [review]
LGTM

**Comment by corona10** [comment]
@erlend-aasland  I will take a look at this PR by this weekend ;) Sorry for the delay!

File: Modules/_sqlite/connection.c [hunk 1]
```diff
@@ -1558,59 +1584,48 @@ pysqlite_connection_iterdump_impl(pysqlite_Connection *self)
     return retval;
 }
 
+/*[clinic input]
+_sqlite3.Connection.backup as pysqlite_connection_backup
+
+    target: object(type='pysqlite_Connection *', subclass_of='pysqlite_ConnectionType') = NULL
+    *
+    pages: int = -1
```
Comment (corona10) [inline Modules/_sqlite/connection.c:1593]: https://docs.python.org/3/library/sqlite3.html?highlight=sqlite3#sqlite3.Connection.backup

Hmm docs looks like wrong.
What do you think?

File: Modules/_sqlite/connection.c [hunk 2]
```diff
@@ -1339,89 +1347,107 @@ PyObject* pysqlite_connection_call(pysqlite_Connection* self, PyObject* args, Py
     return NULL;
 }
 
-PyObject* pysqlite_connection_execute(pysqlite_Connection* self, PyObject* args)
+/*[clinic input]
+_sqlite3.Connection.execute as pysqlite_connection_execute
+
+    sql: unicode
+    /
+    parameters: object = NULL
```
Comment (corona10) [inline Modules/_sqlite/connection.c:1354]: Looks wrong, execute does not accept keyword arguments

**Review by corona10 — CHANGES_REQUESTED** [review]

**Review by corona10 — COMMENTED** [review]
Please check all arguments specs before submitting the patch ;)

File: Modules/_sqlite/connection.c (hunk 2 above)
Comment (erlend-aasland) [reply Modules/_sqlite/connection.c:1354]: Correct. I'll fix.

File: Modules/_sqlite/connection.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/_sqlite/connection.c:1593]: Yes, that's a mistake. I'll fix.

Comment (corona10) [reply Modules/_sqlite/connection.c:1593]: @erlend-aasland 
Yeah but the current migration implementation is right. So I think that we should update the docs.python.org.
How about creating a new issue for this?

Comment (erlend-aasland) [reply Modules/_sqlite/connection.c:1593]: Sounds good. Do you mind opening the issue? I think you can phrase it better than me.

**Comment by erlend-aasland** [comment]
I have made the requested changes; please review again

**Review by corona10 — APPROVED** [review]

**Pull request merged**